src/firebolt/async_db/util.py

```diff
@@ -0,0 +1,36 @@
+from httpx import Response
+
+from firebolt.common.urls import DATABASES_URL, ENGINES_URL
+
+
+async def is_db_available(connection, database) -> bool:
+    """Verify if the database exists"""
+    resp = await _filter_request(
+        connection, DATABASES_URL, {"filter.name_contains": database}
+    )
+    resp.raise_for_status()
+    return len(resp.json()["edges"]) > 0
+
+
+async def is_engine_running(connection, engine) -> bool:
+    """Verify if the engine is running"""
+    resp = await _filter_request(
+        connection,
+        ENGINES_URL,
+        {
+            "filter.name_contains": engine,
+            "filter.current_status_eq": "ENGINE_STATUS_RUNNING",
+        },
+    )
+    resp.raise_for_status()
+    return len(resp.json()["edges"]) > 0
+
+
+async def _filter_request(connection, endpoint: str, filters: dict) -> Response:
```

Comment:
nit: Don't think that we need to pass whole connection here, this could only be `client` and `api_endpoint`. This way we can remove `Connection` dependency for this file
Remove connection from two previous functions if you decide to do so

Reply:
I'd agree if we only used `client` in `_filter_request`, but I feel like `api_endpoint` is not something whatever function or class is checking if engine/db exist should know about.